daemon/src/lib.rs

```diff
@@ -361,3 +365,36 @@ impl Environment {
         }
     }
 }
+
+pub async fn balance_check(
+    margin: Amount,
+    tx_fee_rate: TxFeeRate,
+    build_party_params: &(impl MessageChannel<wallet::BuildPartyParams> + 'static),
+    get_balance: &(impl MessageChannel<wallet::GetBalance> + 'static),
+) -> Result<()> {
+    let (_, pk) = keypair::new(&mut rand::thread_rng());
+    let try_build_params = build_party_params
```

Comment:
❓I guess this is what you meant?

```suggestion
    let build_params = build_party_params
```